packages/core/src/new-api/internal/execution/onchain-state-transitions.ts

```diff
@@ -609,8 +620,19 @@ async function checkTransactionComplete(
     next.txHash
   );
 
+  // No receipt means it hasn't been included in a block yet
   if (receipt === null || receipt === undefined) {
-    // No receipt means it hasn't been included in a block yet
+    if (state.transactionLookupTimer.isTimedOut(next.txHash)) {
```

Comment:
If we get a receipt, we never timeout - even if it hasn't reached the number of confirmations yet.

If the timeout is passed, apply a timeout.